Extension/src/LanguageServer/references.ts

```diff
@@ -254,15 +263,19 @@ export class ProgressHandler {
                                     }
                                     resolve();
                                 } else {
-                                    this.reportProgress(progress, false, isPeek);
+                                    this.reportProgress(progress, false);
                                 }
                             }, this.referencesProgressUpdateInterval);
                         });
                     vscode.window.withProgress(this.referencesProgressOptions, this.referencesProgressMethod);
                     clearInterval(this.referencesDelayProgress);
                 }, this.referencesProgressDelayInterval);
                 break;
+            case ReferencesProgress.CanceledFinalResultsAvailable:
             case ReferencesProgress.FinalResultsAvailable:
+                if (notificationBody.referencesProgress === ReferencesProgress.CanceledFinalResultsAvailable) {
+                    this.referencesCanceled = true;
+                }

```

Comment:
Change to the following?

this.referencesCanceled = notificationBody.referencesProgress === ReferencesProgress.CanceledFinalResultsAvailable;

Reply:
It's possible that might give different results if referencesProgress is FinalResultsAvailable, but this.referencesCanceled got set to true from the UI cancel button.